modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -516,12 +522,16 @@ private List<AbstractNode<?>> localFragments() {
         private CompletableFuture<Void> sendFragment(
                 String targetNodeName, String serialisedFragment, FragmentDescription desc, TxAttributes txAttributes
         ) {
+            Object[] parameterValues = Arrays.stream(ctx.parameters())
+                    .map((v) -> v.value())
+                    .toArray();
```

Comment:
Let's rewrite this avoiding streams and extract this to a method of ArrayUtils class.

Reply:
To reduce number of objects... maybe it will be better to wrap a whole Object[] into DynamicParametersHolder instead of wrapping individual parameters?
```
DynamicParametersHolder {
    Objects[] params;
    BitSet hasValueFlags
}
```

Reply:
I agree, but at the moment there is no separation between prepare and execution and because of that they share the same context class , when in fact for execution we only need exact values ((param_indes, value) ) and for preparation we need (param_index, value/no_value). 

Maybe it would be better to fix this when we introduce a separate preparation phase.

Reply:
Removed java stream usage.

Reply:
I moved code that creates an array of dynamic parameter values from array to `DynamicParameterValue`. Because creating a generic array requires use of reflection to create an array  + such method is very niche.